### ISSUE TITLE
faulty build

### ISSUE THREAD

**Issue by cbontoiu** [ComputationalRadiationPhysics/picongpu#4379]
Hi,

I typically get an executable ```etc/picongpu/bash/mpiexec.tpl``` after building a project but now this is missing and I cannot run. I built the LWFA from the ```picongpu-dev``` with my recent installation (without Spack). The commands were:
```
export PIC_BACKEND="cuda:75" && export OMPI_MCA_io=^ompio
rm -r .build/ && pic-build &> build_log.txt 
```
which seems to be successful (please see the log attached) but execution with this command failed (_The given tpl file "etc/picongpu/bash/mpiexec.tpl" does not exist (-t|--tpl)._
):
```
tbg -s bash -c etc/picongpu/2.cfg -t etc/picongpu/bash/mpiexec.tpl /media/quasar/RawDataDisk2/OUT_TESTS/LaserWakefield
```
Here 2.cfg was modified only to allow 2 GPUs along the y-direction.

Is there something that I miss here?
Many thanks.

Cristian

 
[build_log.txt](https://github.com/ComputationalRadiationPhysics/picongpu/files/10158437/build_log.txt)

**Comment by BrianMarre** [comment]
first:
pic-build did not find openpmd, see
```
Optional Dependencies:
  openPMD: OFF
```
While that is in theory a valid configuration, you will not be able to write out data from your simulation using openPMD, and most of our output is using openPMD, so check that that is really what you want to do.

Second note:
etc/picongpu/bash/mpiexec.tpl is not a executable, rather a configuration file for tbg, that is supposed to be copied from the profile directory from picongpu by `pic-create` as far as I know.
But beware different profiles provide different tpl files, for example the `bash-devServer-hzdr` profile does not provide a `mpiexec.tpl` file.
If a tpl file is missing you may just copy it from a different provided profile, but some changes can be required on some systems.
If you did use the `mpiexec.tpl` files successfully previously, you are probably in luck and do not need to change anything ;)

**Comment by cbontoiu** [comment]
@BrianMarre 

Thanks for your quick reply. I don't use any profile. All dependencies are installed across my machine and they are available. However it is true that I missed ADIOS2 thinking that it is rather used at post processing through Python and Conda.

I will focus on the area marked in this image

![image](https://user-images.githubusercontent.com/7108691/205879013-31a4952d-8ded-4a7f-bfc5-db6221a01ebb.png)

and try to build ADIOS2 from source. This is where I previously used Spack, because installation from the source used to be difficult, two years ago. Hopefully I manage this time.

Kind Regards.
Cristian

**Comment by BrianMarre** [comment]
you should be able to use spack to install openpmd-api independently of picongpu, load it and just use it with picongpu. No need to install it from source. That for example is exactly what the bash-devServer-HZDR install scripts and profiles do.

**Comment by cbontoiu** [comment]
@BrianMarre thanks for your help. Much appreciated. I made some progress and I have now all libraries required. The new log file confirms that openPMD (built with ADIOS2) is in place. I also coppied the LaserWakefield project ```pic-create``` which indeed gives me the ```etc/picongpu/bash/mpiexec.tpl``` configuration file. I will close this issue.

[build_log.txt](https://github.com/ComputationalRadiationPhysics/picongpu/files/10168679/build_log.txt)
